pkg/policy/repository.go

```diff
@@ -379,21 +379,10 @@ func (p *Repository) computePolicyEnforcementAndRules(securityIdentity *identity
 
 	rulesIngress = []*rule{}
 	rulesEgress = []*rule{}
-	// Match cluster-wide rules
-	for rKey := range p.rulesByNamespace[""] {
-		r := p.rules[rKey]
-		if r.matchesSubject(securityIdentity) {
-			if r.Ingress {
-				rulesIngress = append(rulesIngress, r)
-			} else {
-				rulesEgress = append(rulesEgress, r)
-			}
-		}
-	}
-	// Match namespace-specific rules
-	namespace, _ := lbls.LookupLabel(&podNamespaceLabel)
-	if namespace != "" {
-		for rKey := range p.rulesByNamespace[namespace] {
+
+	if subjectIsNode := securityIdentity.ID == identity.ReservedIdentityHost; subjectIsNode {
```

Comment:
Based on https://github.com/cilium/cilium/pull/42580, I think with that in place we will always get the updated node identity in that selector cache - so I'm pretty sure we in that situation are fine to just always rely on the index inside the selector cache without looping through all the rules.